### PR TITLE
bpo-34392: Add sys. _is_interned()

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue34392

https://github.com/python/cpython/issues/78573

### CODE REVIEW

File: Python/sysmodule.c
Comment (pablogsal) [inline Python/sysmodule.c:1005]: Question: Currently `sys.intern`makes a `PyUnicode_CheckExact(s)` over the string so interning a subclass of a string gives a `TypeError` back. This returns `False`. Does it make sense to do a exact check to mirror the interface of `sys.intern`?

Comment (serhiy-storchaka) [reply Python/sysmodule.c:1005]: I don't know.

**Review by methane — APPROVED** [review]

**Comment by csabella** [comment]
@methane, should this be merged after fixing the merge conflicts?  Thanks!

**Comment by methane** [comment]
> @methane, should this be merged after fixing the merge conflicts? Thanks!

Up to @serhiy-storchaka.  He wanted this API to write some tests.

File: Misc/NEWS.d/next/Core and Builtins/2018-08-13-13-25-15.bpo-34392.9kIlMF.rst
Comment (methane) [inline Misc/NEWS.d/next/Core and Builtins/2018-08-13-13-25-15.bpo-34392.9kIlMF.rst]:
```suggestion
Added :func:`sys._is_interned`.
```

**Review by iritkatriel — DISMISSED** [review]
This has merge conflicts now.

**Pull request merged**
